src/java.base/share/classes/jdk/classfile/transforms/CodeLocalsShifter.java

```diff
@@ -40,84 +40,141 @@
 import jdk.classfile.instruction.LocalVariableType;
 
 /**
- *
+ * CodeLocalsShifter is a {@link jdk.classfile.CodeTransform} shifting locals to
+ * new index possitions to avoid conflicts during code injection.
+ * Locals with indexes pointing to the 'this' or method arguments are never shifted.
+ * All locals with indexes pointing beyond the method arguments are re-indexed
+ * in order of appearance.
+ * <p>
+ * A unique index for additional local variable can be manually allocated using
+ * {@link jdk.classfile.transforms.CodeLocalsShifter#addLocal(jdk.classfile.TypeKind)}
+ * at any time before, during, or after the transformation.
+ * <p>
+ * A fork of the actual CodeLocalsShifter can be obtained from
+ * {@link jdk.classfile.transforms.CodeLocalsShifter#fork()}.
+ * Forked CodeLocalsShifter starts with clean locals mapping, however it respects original
+ * method argument locals and avoids conflicts with previously allocated or shifted locals.
+ * Forked CodeLocalsShifter does not dynamically sync with its parent, nor vice versa.
+ * <p>
+ * Sample use in complex transformation:
+ * <p>
+ * {@snippet lang=java :
+ *     var localsShifter = new CodeLocalsShifter(methodModel.flags(), methodModel.methodTypeSymbol());
```

Comment:
Should this be `CodeLocalsShifter.of(` rather than `new CodeLocalsShifter`?

Reply:
yes, fixed, thanks!